### ISSUE TITLE
demultiplex scATAC data by sex

### ISSUE THREAD

**Issue by Bassi-git** [nkschaefer/cellbouncer#4]
Hi,
Thank you for this fantastic tool, I've successfully used it to demultiplex by species without any issues!

I’m now exploring whether it might be feasible to adapt one of your pipelines to demultiplex single-cell ATAC-seq data based on sex. Unfortunately, we don’t have much genomic information for the individuals, so demultiplexing by individual isn't ideal in our case.

Would you have any recommendations or thoughts on whether this could work with your pipeline, or how we might approach it?

Thanks again for your work on this tool!

**Comment by nkschaefer** [comment]
Hey,

Sorry for the delay, and thanks for using CellBouncer!

That's a good question -- this might be easier to do within an analysis toolkit like scanpy, and I'm probably not the most qualified to give advice here (I've mostly analyzed scATAC-seq in the context of paired scRNA-seq). But you might be able to fit a mixture model on counts of informative ATAC peaks. I would expect the promoter of SRY to have a male-specific peak and the promoter of XIST to have a female-specific one. There are probably plenty of other peaks to look at, but I imagine they depend on cell type, developmental time, etc. and other stuff I'm less aware of.

Because of the sparsity of the data, I'd first try imputing counts using a tool like [MAGIC](https://github.com/KrishnaswamyLab/MAGIC) and see if there are peaks in those regions with bimodal (imputed) per-cell count distributions. If there are, you could identify the cells by fitting a mixture model to the count distributions for those peaks. You might be able to use a tool like [pomegranate](https://pomegranate.readthedocs.io/en/latest/index.html) for this - I've used an old version and thought it was convenient when working in Python. I would expect there to still be unidentifiable cells because of data sparsity, but if this happens, might be able to use something like [K-nearest neighbors classification](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsClassifier.html) to borrow sex labels from similar cells. 

As far as CellBouncer is concerned, `demux_species` can take in a text matrix, where rows are cell barcodes and columns are features (these are normally counts of k-mers specific to each species, but in your case, they could be counts of reads overlapping the TSS of a sex-informative gene like SRY or XIST). If you wanted to include multiple male-biased and/or female-biased ATAC peaks, you would need to sum counts for all male-biased peaks per cell into one number and then sum all female-biased peak counts per cell into another. If you can extract those data, you can put them in a tab-separated text file called `species_counts.txt` (with no column headers), and then create a second file called `species_names.txt`. This file (`species_names.txt`) should just have two rows saying "[index][tab]male" and "[index][tab]female", in the order matching the columns in `species_counts.txt` (e.g. if column 1 is read counts at the SRY promoter and column 2 is read counts at the XIST promoter, the file would look like this):
```
0    male
1    female
```
Or if the order is flipped (e.g. XIST counts followed by SRY counts):
```
0    female
1    male
```
You would then put both `species_counts.txt` and `species_names.txt` into a directory named whatever you like, and then run `demux_species -o [directory_name]` from outside that directory. You could then run `plot/demux_species.R` on the results to see how well that worked.

I'd start by plotting the counts, though, since I'm not sure how feasible this will be, given typically low counts of ATAC reads per cell. Hopefully the imputation helps, or you might be able to find a set of tightly correlated peaks that are all clearly sex-biased, and you could sum counts from those across cells.

Hope this helps!

**Comment by Bassi-git** [comment]
Hi,
Thank you very much for all your suggestions! This definitely already helps. 
I will look into it and hopefully come up with a solution.
